Test failures for Linux 32-bit

I've built a test grid that builds Pillow 3.2.0 wheels for manylinux1, and tests them : 

* https://travis-ci.org/matthew-brett/pillow-wheels/builds/139079195
* https://github.com/matthew-brett/pillow-wheels/tree/refactor-to-multibuild

In particular, I'm testing the 32-bit builds on 32-bit docker images.  The 32-bit builds are all failing with a single test failure:

https://travis-ci.org/matthew-brett/pillow-wheels/jobs/139079199#L1113

```
======================================================================
FAIL: TestImageCms.test_extended_information
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/io/Pillow/Tests/test_imagecms.py", line 262, in test_extended_information
    self.assertEqual(p.attributes, 4294967296)
AssertionError: -4587544 != 4294967296L
```

Any ideas what might be going on?

Testing the fix in #1975, I'm also getting precision errors with: p.blue_colorant, p.chromatic_adaption, p.green_primary, p.media_black_point, p.media_white_point and p.media_white_point_temperature. 

These are pretty much all small differences in the least significant digit: e.g.:

```
======================================================================
FAIL: TestImageCms.test_extended_information
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Pillow/Tests/test_imagecms.py", line 290, in test_extended_information
    self.assertEqual(p.media_white_point, ((0.964202880859375, 1.0, 0.8249053955078125), (0.3457029219802284, 0.3585375327567059, 1.0)))
AssertionError: Tuples differ: ((0.964202880859375, 1.0, 0.82... != ((0.964202880859375, 1.0, 0.82...

First differing element 1:
(0.34570292198022834, 0.3585375327567059, 1.0)
(0.3457029219802284, 0.3585375327567059, 1.0)

  ((0.964202880859375, 1.0, 0.8249053955078125),
-  (0.34570292198022834, 0.3585375327567059, 1.0))
?                    -

+  (0.3457029219802284, 0.3585375327567059, 1.0))

----------------------------------------------------------------------
Ran 19 tests in 0.431s
```

--- a/_imagingcms.c
+++ b/_imagingcms.c
@@ -983,12 +983,9 @@ cms_profile_getattr_attributes(CmsProfileObject* self, void* closure)
 {
     cmsUInt64Number attr;
     cmsGetHeaderAttributes(self->profile, &attr);
-#ifdef _WIN32
-    // Windows is weird this way.
-    return PyLong_FromLongLong((long long) attr);
-#else
-    return PyInt_FromLong((long) attr);
-#endif
+    /* This works just as well on Windows (LLP64), 32-bit Linux
+       (ILP32) and 64-bit Linux (LP64) systems.  */
+    return PyLong_FromUnsignedLongLong((unsigned long long) attr);
 }
 
 static PyObject*

